### ISSUE TITLE
SharePoint webpart settings are reset when editing tab in new MS Teams clients

### ISSUE THREAD

**Issue by xxyziggy** [SharePoint/sp-dev-docs#9464]
### Target SharePoint environment

SharePoint Online

### What SharePoint development model, framework, SDK or API is this about?

💥 SharePoint Framework

### Developer environment

None

### What browser(s) / client(s) have you tested

- [ ] 💥 Internet Explorer
- [X] 💥 Microsoft Edge
- [X] 💥 Google Chrome
- [ ] 💥 FireFox
- [ ] 💥 Safari
- [ ] mobile (iOS/iPadOS)
- [ ] mobile (Android)
- [ ] not applicable
- [X] other (enter in the "Additional environment details" area below)

### Additional environment details

**Microsoft Teams Desktop Client:**
Microsoft Teams version: 23320.3027.2591.1505
Client version: 49/23113028722
Last updated: 16.01.2024
Operating System: Windows 11

**Microsoft Teams Web Client:**
Microsoft Teams version: [Provide the version number for the web client, if available.]
Client version: 415/23113028722

### Describe the bug / error

SharePoint webpart settings in MS Teams tab are reset when editing tab in new MS Teams clients

### Steps to reproduce

1. Create a Teams app with SharePoint webpart in it, webpart must have configurable properties
2. In app manifest use
```
 "configurableTabs": [
        {
            "canUpdateConfiguration": true,
            "configurationUrl": "https://{teamSiteDomain}{teamSitePath}/_layouts/15/TeamsLogon.aspx?SPFX=true&dest={teamSitePath}/_layouts/15/teamshostedapp.aspx%3FopenPropertyPane=true%26teams%26componentId=17abc1ef-582b-4ced-96fe-a5621de1f5ec%26forceLocale={locale}",
            "scopes": ["team"],
            "context": ["channelTab"]
        }
    ],
```
3. Publish the app, and create a Tab in MS Teams with this webpart and confugure its properties
4. Click Tab settings
5. Popup opens, after clicking Save button webpart props panel is opened, but properties are empty and webpart state goe back to unconfigured state

![new teams](https://github.com/SharePoint/sp-dev-docs/assets/1702446/c510786b-a570-4de3-82cf-36ed2a5bf015)

### Expected behavior

In old teams clients after clicking Tab settins no popup appears, webpart configuration panel opens with saved properties
![old teams](https://github.com/SharePoint/sp-dev-docs/assets/1702446/d5459150-9af3-4c30-9ab8-a061ab9ce322)

**Comment by ruiAfonsoISEP** [comment]
This issue is also occurring in my environment with custom SPFx web part. The correct experience is functioning in the old Microsoft Teams desktopp app.

**Comment by UeliSchwarz** [comment]
I observe the same behavior, settings in teams get lost on a roundtrip.

**Comment by antonkhrit** [comment]
We experience the same bug in Teams (NEW). SPFx properties are reset every time we open settings.

**Comment by shaipetel** [comment]
Just had several very unhappy customers reporting that all their settings got lost when they try to edit the tab properties.
There used to be an "edit" button, now there is only "settings" and it opens a popup and deletes all properties.

Any updates on this?

**Comment by shaipetel** [comment]
I think this is a duplicate of this issue [9389](https://github.com/SharePoint/sp-dev-docs/issues/9389)

**Comment by kandharithirumalreddy** [comment]
Hi,

we have facing the same issue in New Teams. Any update?

FYI : we have upgraded the solution to SPFx 1.18.2. 

Already reported here as well [](https://github.com/SharePoint/sp-dev-docs/issues/9389)

**Comment by singingknight** [comment]
I agree that this is a very annoying bug that makes SPFX web parts unusable as Teams Tabs.

But I don't agree with the described expected behavior. I would prefer that the behavior was that when adding/updating the tab, the following should happen:

  1. The dialog should appear, but with the property pane instead of just a Teams logo
  2. When save was clicked then settings was saved to HostedAppConfigs and dialog dismissed
  3. Then the tab should load webpart with specified properties and no property pane.

**Comment by kandharithirumalreddy** [comment]
Hi,

any update on this issue, when will it be fixed? will the fix be part of next release?

Now Microsoft forcing to use New Teams. So it's effecting our customers. Now It's become major issue for us.

**Comment by kandharithirumalreddy** [comment]
Hi,

any update on this issue, when will it be fixed? will the fix be part of next release?

Now Microsoft forcing to use New Teams. So it's effecting our customers. Now It's become major issue for us

**Comment by kandharithirumalreddy** [comment]
Hi,

any update on this issue, when will it be fixed? will the fix be part of next release?

Now Microsoft forcing to use New Teams. So it's effecting our customers. Now It's become major issue for us.

It's strange that no updates from the assigned persons

**Comment by AJIXuMuK** [comment]
Sorry for the long delay folks.
We have identified the issue and will be working with Microsoft Teams partners to fix it.
Will update the thread when we have rollout ETA or other updates.

**Comment by PMcGown** [comment]
Thank you for the update, @AJIXuMuK

**Comment by SandyLT** [comment]
Thank you, @AJIXuMuK ! We (and our customers) will look forward to this fix! For us, it's especially impacting our Social Squared discussion board app, because many of our clients use it in Teams in addition to SharePoint, and Team Owners use the Settings panel for a lot of configuration.

**Comment by AJIXuMuK** [comment]
So the issue is being worked on.
We are expecting the fix to be rolled out globally by mid August.

**Comment by PMcGown** [comment]
Thanks for the update, @AJIXuMuK . We appreciate you prioritizing this issue.

**Comment by kandharithirumalreddy** [comment]
Hi @AJIXuMuK 

Any update on this fix to be rolled ?

Thanks
Thirumal Reddy

**Comment by AJIXuMuK** [comment]
@kandharithirumalreddy - one of the fixes has rolled out but it is still working not ideally. We're working on additional fixes.

Good thing is - the properties are not reset anymore.
Bad thing - settings pane is not always opening.

You may need to reload a tab multiple times to open the pane.
See the video:

https://github.com/user-attachments/assets/26e7f69c-a3e3-47e6-bf34-03882c025038

**Comment by PMcGown** [comment]
Thanks for the update, @AJIXuMuK

**Comment by kandharithirumalreddy** [comment]
Hi @AJIXuMuK 

That's great.

Thank you very much for your update.

We will try to test internally.

Thanks
Thirumal Reddy

**Comment by Amey-MSFT** [comment]
Hello @xxyziggy,
Thank you for bringing this issue to our attention. Could you please confirm if the issue still persists for you?

**Comment by xxyziggy** [comment]
@Amey-MSFT it's working now, thanks for the update!
